src/contracts/services/derivativeShd.ts

```diff
@@ -0,0 +1,114 @@
+import { getActiveQueryClient$ } from '~/client';
+import {
+  switchMap,
+  first,
+  map,
+  lastValueFrom,
+} from 'rxjs';
+import { sendSecretClientContractQuery$ } from '~/client/services/clientServices';
+import {
+  FeeResponse,
+  ContractStatusResponse,
+  StatusLevel,
+  StakingInfoResponse,
+} from '~/types/contracts/derivativeShd/response';
+import {
+  ParsedFeeResponse,
+  ParsedStakingInfoResponse,
+} from '~/types/contracts/derivativeShd/model';
+import { convertCoinFromUDenom } from '~/lib/utils';
+import { msgQueryStakingInfo } from '~/contracts/definitions/derivativeShd';
+
+// Contract returns price as a rate of dSHD/SHD with 6 decimals
+const DERIVATE_PRICE_DECIMALS = 6;
+
+/**
+* Parses the contract fee query into a cleaner data model
+*/
+const parseDerivativeShdFees = (response: FeeResponse): ParsedFeeResponse => ({
+  stakingFee: response.staking.rate / (10 ** response.staking.decimal_places),
+  unbondingFee: response.unbonding.rate / (
+    10 ** response.unbonding.decimal_places
+  ),
+  feeCollector: response.collector,
+});
+
+/**
+* Parses the contract status query into a cleaner data model
+*/
+const parseDerivativeShdContractStatus = (
```

Comment:
is this parser unused because it's built into staking info? Let's delete it.

Reply:
Done